### PR TITLE
Update entitlement to add Course so the entitlement can be matched to…

### PR DESCRIPTION
**Pull request by mcginkel**
Update entitlement to add Course so the entitlement can be matched to SIS data.
Adjusted documentation in provider to match sample description
Fixed security definition

### CODE REVIEW

File: reference/entitlement.v1.yaml [hunk 1]
```diff
@@ -304,14 +304,20 @@ components:
...
         subjectCode:
           type: string
-          description: Government standard subject code.
+          description: 'Government standard subject code. bron: Ministerie van OCW, like https://zoek.officielebekendmakingen.nl/stcrt-2019-35044'
         level:
           type: string
-          description: Government standard level - e.g. havo-1
+          description: 'Official Level, followed by courseyear. example ''havo-1, vwo-2, etc.'''
+        course:
+          type: string
```
Comment (niesink) [inline reference/entitlement.v1.yaml:316]: I can see some drawbacks to using the name of the course to match/group entitlements to SIS data, as this name could be altered by the school at any time, which doesn't seem unimaginable, especially during the summer. 

Would it perhaps be a safer option to use either a SIS-generated uuid or [the government code of the course](https://wetten.overheid.nl/jci1.3:c:BWBR0043849&bijlage=I&z=2020-08-01&g=2020-08-01) for this? 

The advantage of this being that they can't be altered (as easily) by the school. 

The advantage of the SIS-generated uuid over the government code is that the school can differentiate between two of their courses that are both connected to the same official/national course (a Technasium and a non-Technasium instance of Havo Natuur & Techniek for example).

This field would have to be part of the SIS API as well if we want to use it here.

Comment (cliftonc) [reply reference/entitlement.v1.yaml:316]: Agree - it feels safer to have `courseCode` here, and if necessary the friendly name the school uses to describe it if it helps everyone understand what it actually means in each system.

Comment (mcginkel) [reply reference/entitlement.v1.yaml:316]: Agreed, Then I will update it in the SIS-data-api as well. The courseCode I see in the system are self describing for the teachers and do not change during the year. Can we start with just the code? I think that will be enough to start with.

Comment (mcginkel) [reply reference/entitlement.v1.yaml:316]: I think the "government code for the course" is what we place in "Level". So independent how the school calls it we have the official name/code. Until now I specified "Havo-1" in the samples, we could use "0300" and each system can have a lookuptabla if needed.
In this way we have the name that the schools calls the course and the code how the Publishers/LA calls the course. We need both.

If we want to use the government code, then we are dependent on the school SIS to supply this data, and I know that that most schools use this, but it is not 100% available. so It should still be optional. This makes it not always possible  to present the best possible match of products to courses while we create LML's but it is workable.

Comment (niesink) [reply reference/entitlement.v1.yaml:316]: Ok, I'm not sure using any string that can be changed by the school is safe, but you're probably right that it won't change very often during the schoolyear, so we can use the courseCode to allow parties to logically group certain students (and entitlements) together.

As for Level, I'm not sure using the government code of the course makes much sense in that context. A student doesn't take a specific subject at the level 'Gymnasium Economie & Maatschappij', they take the subject at the level 'VWO year 5'.

If MP/LA/LMS need to know whether a student is taking the course 'Gymnasium Economie & Maatschappij' or 'Atheneum Natuur & Techniek' then I think this is something that should be included in the SIS API, but not as part of the subject choice. The object we're talking about in that case is the 'enrollment'.

Comment (edwinverwoerd) [reply reference/entitlement.v1.yaml:316]: I think we have to go futher with this and say yes to this request.

About Level please let use official codes unless we want to use or we will create our own list of definitions. It is not a good idea to ad  a free text field. The value occurs in multiple api (transactions).

Comment (edwinverwoerd) [reply reference/entitlement.v1.yaml:316]: Lets have a discussion about this later in, mayby after the first phase..

Comment (mcginkel) [reply reference/entitlement.v1.yaml:316]: Adding the information here was to get in sync with the course/level/subject definition in the SIS data, so we can match.
If we have this discussion on how to fill course or level that is a valid discussion, but can we move that to the sis data, so we can get that consistent over the whole API (SIS, Cataloq, usage, entitlements) ?
And approve this one to get consistent ?

Comment (cliftonc) [reply reference/entitlement.v1.yaml:316]: Action to close:  First of all ensure that the definition between SIS Api and Entitlement Api are consistent; later review if the approach of the course related data is correct across all services (after prototype).

**Review by cliftonc — APPROVED** [review]
Reviewed in mtg

**Pull request merged**